hrjobcontract/js/src/job-contract/services/utils.service.js

```diff
@@ -72,6 +73,40 @@ define([
         return deffered.promise;
       },
 
+      getOptionValues: function (groupNames) {
+        var deferred = $q.defer();
+        var multiple = _.isArray(groupNames);
+
+        CRM.api3('OptionValue', 'get', {
+          'sequential': 1,
+          'is_active': 1,
+          'option_group_id.name': { 'IN': multiple ? groupNames : [groupNames] },
+          'return': [ 'id', 'label', 'weight', 'value', 'is_active', 'option_group_id', 'option_group_id.name' ],
+          'options': {
+            'limit': 1000,
+            'sort': 'id'
+          }
+        })
+          .done(function (result) {
```

Comment:
why do you need to manipulate the response? What's the format that you get from the api endpoint?

Reply:
This was done to make it conform with implementation in job roles and affords having the list of option groups returned as an object. The whole implementation will be replace with API.resource. 

A similar [implementation](https://github.com/compucorp/civihr/blob/master/hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.controller.js#L379) to API.resource using `OptionGroup.valuesOf()` was used initially but this was throwing errors. That is why i decided to use CRM.api3.

![screen shot 2018-11-15 at 10 19 33 am](https://user-images.githubusercontent.com/1507645/48558407-80c1ab00-e8e9-11e8-9ede-6ea5b53f8dd8.png)